pycbc/events/stat.py

```diff
@@ -2107,12 +2133,28 @@ def lognoiserate(self, trigs):
 
         Returns
         ---------
-        lognoisel: numpy.array
+        lognoiserate: numpy.array
             Array of log noise rate density for each input trigger.
         """
+
+        # make sure every trig has a dq state
+
+        try:
+            # works in offline
+            ifo = trigs.ifo
+        except AttributeError:
+            ifo = trigs['ifo']
+            unq = numpy.unique(ifo)
+            assert len(unq) == 1
+            # Should be exactly one ifo provided
+            ifo = unq[0]
+
+        dq_state = self.find_dq_state_by_time(ifo, trigs['end_time'][:])
+        dq_rate = self.find_dq_noise_rate(trigs, dq_state)
+
         logr_n = ExpFitFgBgNormStatistic.lognoiserate(
                     self, trigs)
-        logr_n += self.find_dq_val(trigs)
+        logr_n += numpy.log(dq_rate)
```

Comment:
When calculating the log of the noiserate for a trigger, the code determines the dq state of each trigger (which set of segments it fell into out of clean, dq_flag, or gated) and then looks up the appropriate trigger rate adjustment, which depends on the triggers' ifo, template, and dq state